lib/crypto/index.js

```diff
@@ -89,32 +89,47 @@ function Crypto(baseApis, eventEmitter, sessionStore, userId, deviceId) {
 
 function _registerEventHandlers(crypto, eventEmitter) {
     eventEmitter.on("sync", function(syncState, oldState, data) {
-        if (syncState == "PREPARED") {
-            // XXX ugh. we're assuming the eventEmitter is a MatrixClient.
-            // how can we avoid doing so?
-            var rooms = eventEmitter.getRooms();
-            crypto._onInitialSyncCompleted(rooms);
+        try {
+            if (syncState == "PREPARED") {
+                // XXX ugh. we're assuming the eventEmitter is a MatrixClient.
+                // how can we avoid doing so?
+                var rooms = eventEmitter.getRooms();
+                crypto._onInitialSyncCompleted(rooms);
+            }
+        } catch (e) {
+            console.error("Error handling sync", e);
         }
     });
 
-    eventEmitter.on(
-        "RoomMember.membership",
-        crypto._onRoomMembership.bind(crypto)
-    );
+    eventEmitter.on("RoomMember.membership", function(event, member, oldMembership) {
+        try {
+            crypto._onRoomMembership(event, member, oldMembership);
+        } catch (e) {
+             console.error("Error handling membership change:", e);
```

Comment:
Should I try and include information about the RoomID and UserID or the EventID here?